### PR TITLE
feat: added feature to upload chart to ECR

### PR DESCRIPTION
**Pull request by junydania**
**Fixes or implements IN-1055**

### Brief description. What is this change?

Refactor helm publish action to push to ECR

### Related PRs

- https://github.com/voiceflow/XXXXXXXXX/pull/123

### Checklist

- [ ] Breaking changes have been communicated, including:
    - New required environment variables
    - Renaming of interfaces (API routes, request/response interface, etc)
- [ ] New environment variables have [been deployed](https://www.notion.so/voiceflow/Add-Environment-Variables-be1b0136479f45f1adece7995a7adbfb)
- [ ] Appropriate tests have been written
    - Bug fixes are accompanied by an updated or new test
    - New features are accompanied by a new test

### CODE REVIEW

File: src/jobs/helm/helm-publish-charts-ecr.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  - checkout_shallow_clone
+  - steps: << parameters.prepublish_steps >>
+  - helm-add-repos
+  - run: &package-publish-charts-ecr
+      name: Package and publish charts
+      working_directory: << parameters.chart_directory >>
+      command: <<include(scripts/helm/package-publish-charts-ecr.sh)>>
+      environment:
+        CHARTS: << parameters.charts >>
+  - run: *package-publish-charts-ecr
```
Comment (xavidop) [inline src/jobs/helm/helm-publish-charts-ecr.yaml:25]: why is it executed twice?

**Review by xavidop — CHANGES_REQUESTED** [review]

File: src/scripts/helm/publish-prerelease-ecr.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+echo "ECR_REPOSITORY_URI: ${ECR_REPOSITORY_URI}"
+
+# Enable experimental OCI support in Helm
+export HELM_EXPERIMENTAL_OCI=1
+
+# Login to ECR
+aws ecr get-login-password --region "$AWS_REGION" | helm registry login --username AWS --password-stdin "$ECR_REPOSITORY_URI"
+
+# Publishes beta releases for helm charts in subdirectories of working directory
+for file in * ; do
```
Comment (edison-vflow) [inline src/scripts/helm/publish-prerelease-ecr.sh:19]: Why is the logic for releases and pre-releases different.
For charts we loop over `for chart in ${CHARTS?}; do` and for pre-releases we loop over `for file in * ; do`
Is not the fundamental difference that the chart repo is different but the release logic would be close to similar as well ?
I dont know much here but just wondering why a change of the repo would change the way we also package the charts

File: src/jobs/helm/helm-publish-charts-ecr.yaml (hunk 1 above)
Comment (junydania) [reply src/jobs/helm/helm-publish-charts-ecr.yaml:25]: Fixed

File: src/scripts/helm/publish-prerelease-ecr.sh (hunk 2 above)
Comment (junydania) [reply src/scripts/helm/publish-prerelease-ecr.sh:19]: Good question.. The logic seems the same but just probably naming I guess

**Review by xavidop — APPROVED** [review]

**Pull request merged**